# Refining the taxonomy of the order *Hyphomicrobiales* (*Rhizobiales*) based on whole genome comparisons of over 130 type strains

diCenzo GC, Yuqi Y, Young JPW, KuzmanoviÄ‡ N. 2023. **Refining the taxonomy of the order *Hyphomicrobiales* (*Rhizobiales*) based on whole genome comparisons of over 130 genus type strains** *bioRxiv*. [https://doi.org/10.1101/2023.11.15.567303]

This repository contains the scripts and input files required to repeat the analyses reported in the associated article. In addition, this repository contains the sets of marker proteins used for phylogenetic analyses and calculation of core-proteome average amino acid identity, which can be used with the provided pipeline to classify organisms at the family level in the order *Hyphomicrobiales*.

## 1_Rhizobiales

This folder contains the overall workflow (workflow.sh), accessory scripts, and input files required to repeat the genome-based phylogenetic reconstructions, average amino acid identity calculations, and 16S rRNA gene phylogenetic analyses.

## 2_Phyllobacteriaceae

This folder contains the overall workflow (workflow.sh), accessory scripts, and input files required to repeat the genome-based phylogenetic reconstructions focused on the family *Bartonellaceae* and related families.

## 3_Pipelines

This folder contains a pipeline and, with instructions, for using the taxonomic framework proposed in our study to assign new genera to families within the order *Hyphomicrobiales*. It additionally includes the protein sequences and protein alignments of each of the marker proteins used in our study, which are required for using the provided pipeline.